internal/pkg/manifest/workload.go

```diff
@@ -25,25 +26,47 @@ import (
 	"gopkg.in/yaml.v3"
 )
 
+// AWS VPC subnet placement options.
 const (
-	defaultFluentbitImage = "amazon/aws-for-fluent-bit:latest"
-	defaultDockerfileName = "Dockerfile"
-
-	// AWS VPC subnet placement options.
 	PublicSubnetPlacement  = "public"
 	PrivateSubnetPlacement = "private"
 )
 
+// Platform options.
+const (
+	OSLinux                 = dockerengine.OSLinux
+	OSWindows               = dockerengine.OSWindows
+	OSWindowsServer2019Core = "windows_server_2019_core"
+	OSWindowsServer2019Full = "windows_server_2019_full"
+
+	ArchAMD64 = dockerengine.ArchAMD64
+	ArchX86   = dockerengine.ArchX86
+)
+
+const (
+	defaultFluentbitImage = "amazon/aws-for-fluent-bit:latest"
+	defaultDockerfileName = "Dockerfile"
+)
+
 var (
 	// WorkloadTypes holds all workload manifest types.
 	WorkloadTypes = append(ServiceTypes, JobTypes...)
 
 	// All placement options.
 	subnetPlacements = []string{PublicSubnetPlacement, PrivateSubnetPlacement}
 
-	validPlatforms        = []string{dockerengine.DockerBuildPlatform(dockerengine.LinuxOS, dockerengine.Amd64Arch)}
-	validOperatingSystems = []string{dockerengine.LinuxOS}
-	validArchitectures    = []string{dockerengine.Amd64Arch}
+	validShortPlatforms = []string{
+		dockerengine.PlatformString(OSLinux, ArchAMD64),
+		dockerengine.PlatformString(OSLinux, ArchX86),
+		dockerengine.PlatformString(OSWindows, ArchAMD64),
+		dockerengine.PlatformString(OSWindows, ArchX86),
+	}
+	validAdvancedPlatforms = []PlatformArgs{
+		{OSFamily: aws.String(OSLinux), Arch: aws.String(ArchAMD64)},
+		{OSFamily: aws.String(OSLinux), Arch: aws.String(ArchX86)},
+		{OSFamily: aws.String(OSWindowsServer2019Core), Arch: aws.String(ArchX86)},
+		{OSFamily: aws.String(OSWindowsServer2019Full), Arch: aws.String(ArchX86)},
```

Comment:
What about 
```
		{OSFamily: aws.String(OSWindowsServer2019Core), Arch: aws.String(ArchAMD64)},
		{OSFamily: aws.String(OSWindowsServer2019Full), Arch: aws.String(ArchAMD64)},
```
?

Reply:
I wasn't sure! 😄   
I stuck to only `x86_64` to match the API, but we do allow `amd64` for linux so we can make the same argument here. 

I updated it!